runtime/lua/vim/_defaults.lua

```diff
@@ -311,4 +312,108 @@ do
       end
     end)
   end
+
+  --- If the TUI was unable to determine that the host terminal supports truecolor, disable
+  --- 'termguicolors' and query the terminal. If the terminal's response indicates that it does
+  --- support truecolor, re-enable 'termguicolors'.
+  do
+
+    --- Set 'termguicolors', but only if it was not already set by the user.
+    ---
+    --- @param enable boolean If true, enable 'termguicolors', else disable
+    local function settgc(enable)
+      if not vim.api.nvim_get_option_info2('termguicolors', {}).was_set then
+        vim.o.termguicolors = enable
```

Comment:
Looks like `settgc` is only ever called with `true`. Also, it looks possible that `settgc` is called twice, but the 2nd call will be ignored, is that intentional?

Reply:
`settgc` can be called in indeterminate order, depending on when responses from the terminal arrive. We only want to set it if not already set by the user, which is why it's guarded.

But this is broken right now anyway, see my [earlier comment](https://github.com/neovim/neovim/pull/26407#issuecomment-1841743997). I need to think about this some more.